### PR TITLE
sign-up page added

### PR DESCRIPTION
**Pull request by pariyar07**

### CODE REVIEW

File: js/sign-up.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    formControl.className = "form-control error"
+}
+
+function setSuccessFor(input) {
+    const formControl = input.parentElement;
+    formControl.className = "form-control success"
+}
+
+function isEmail(email) {
+    return /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(email);
```
Comment (sanjitsarkar) [inline js/sign-up.js:62]: you can use input type to email to make the form to validate automatically

File: styles/nav.css [hunk 2]
```diff
@@ -1,69 +1,89 @@
...
+    border-style: solid;
+    border-image:
+        linear-gradient(to right,
+            red,
+            rgba(0, 0, 0, 0))0 1 100%;
 }
-.nav-title{
+
+.nav-title {
     background: linear-gradient(to right, #2b2b2b 0%, #ffffff 95%);
```
Comment (sanjitsarkar) [inline styles/nav.css:16]: use color variable

**Review by sanjitsarkar — APPROVED** [review]

File: js/sign-up.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        setSuccessFor(password)
+    }
+
+    if(checkPasswordValue === "") {
+        setErrorFor(checkPassword, "Please enter a same password");
+    } else if(checkPasswordValue !== passwordValue) {
+        setErrorFor(checkPassword, "Password does not match");
+    } else {
+        setSuccessFor(checkPassword)
+    }
```
Comment (saurabhchirde) [inline js/sign-up.js:46]: no need to use js for this, you can directly make these input fields as 'required 'in HTML itself.

File: styles/nav.css [hunk 4]
```diff
@@ -1,69 +1,89 @@
...
+.badge-cart-count {
     display: flex;
     align-items: center;
     justify-content: center;
     position: absolute;
     right: -8px;
     top: -5px;
     background: var(--badge-bg);
-    color:rgb(255, 255, 255);
+    color: rgb(255, 255, 255);
```
Comment (saurabhchirde) [inline styles/nav.css:42]: Try to use color variables unless you are not modifying the "alpha" channel.

File: styles/nav.css [hunk 5]
```diff
@@ -1,69 +1,89 @@
...
 }
-.nav-list>li>p{
+
+.nav-list>li>p {
     font-size: 0.8rem;
 }
 
-.nav-container-search{
+.nav-container-search {
     border: 1px solid rgb(150, 150, 150);
```
Comment (saurabhchirde) [inline styles/nav.css:61]: color variables.

**Review by saurabhchirde — APPROVED** [review]
The code looks good to merge with a few changes.
Just a suggestion, to not use such heavy IMG files. Try to use compressed.

File: js/sign-up.js (hunk 3 above)
Comment (pariyar07) [reply js/sign-up.js:46]: This code was a practise code I had done before the projects. Just implemented it.

**Pull request merged**